### DISCUSSION TITLE
How old is the topological map used by the XCSoar Map generator ?

### DISCUSSION THREAD

**Discussion by sasam** [XCSoar/XCSoar#1564] [General]
Some roads that were put into traffic almost 10 years ago are not present in the map.

**Comment by lordfolken** [comment]
The osm export is from this year. Care to point to a road that is missing on https://openstreetmap.org ?

You will find the filter scripts here:
https://github.com/XCSoar/xcsoar-mapgen-topology/tree/master/scripts

You can try with an extract from your region: http://download.geofabrik.de/

**Comment by sasam** [comment]
The roads I noticed were missing exist in OSM.

In the attached picture, I marked the missing roads in yellow (It is near Zagreb, Croatia). 
![misssing_road](https://github.com/user-attachments/assets/f3f22d63-b8e5-4f9b-8118-cb5862062e57)

**Comment by lordfolken** [comment]
https://www.openstreetmap.org/way/95093006

So according to the tags in OSM this is a highway = trunk

Unfortunatly this get's currently filtered at the reduction stage of the topography:
https://github.com/XCSoar/xcsoar-mapgen-topology/blob/860ea26e5190aa369282cc7480bd461e7a4d18eb/scripts/reduce_roads_big.py#L28C1-L28C90

I updated the script to include trunks and trunk_links. 
https://github.com/XCSoar/xcsoar-mapgen-topology/blob/master/scripts/reduce_roads_big.py

 I don't know when i will be able to produce new topography. It needs quite a bit of hardware (128GB Ram 3TB disk..)

**Comment by lordfolken** [comment]
https://github.com/XCSoar/xcsoar-mapgen-topology/issues/16
